ouroboros-network/src/Cardano/Node/PublicRootPeers.hs

```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE NamedFieldPuns #-}
+
+module Cardano.Node.PublicRootPeers where
+
+import Data.Map.Strict (Map)
+import Data.Map.Strict qualified as Map
+import Data.Set (Set)
+import Data.Set qualified as Set
+import Data.Typeable (Typeable)
+import Ouroboros.Network.PeerSelection.PeerAdvertise (PeerAdvertise)
+
+data CardanoPublicRootPeers peeraddr =
+  CardanoPublicRootPeers
+    { cprpGetPublicConfigPeers :: !(Map peeraddr PeerAdvertise)
+    , cprpGetBootstrapPeers    :: !(Set peeraddr)
+    }
+  deriving (Eq, Show, Typeable)
```

Comment:
This type should be called `CardanoExtraPeers`, and then we could provide & use the following:
```hs
type CardanoPublicRootPeers peeraddr = PublicRootPeers (CardanoExtraPeers peeraddr) peeraddr 
```

ref: bae4f75ae3967ea2dcff75b569eb6102a0a2d928